File: C/heapsort/llama-3.3-70b-versatile/llama-3.3-70b-versatile.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Function to swap two elements
void swap(int* a, int* b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

// Function to heapify a subtree rooted at index
void heapify(int arr[], int n, int index) {
    int largest = index;
    int left = 2 * index + 1;
    int right = 2 * index + 2;

    if (left < n && arr[left] > arr[largest])
        largest = left;

    if (right < n && arr[right] > arr[largest])
        largest = right;

    if (largest != index) {
        swap(&arr[index], &arr[largest]);
        heapify(arr, n, largest);
    }
}

// Function to implement heap sort
void heapSort(int arr[], int n) {
    // Build a max heap
    for (int i = n / 2 - 1; i >= 0; i--)
        heapify(arr, n, i);

    // Extract elements one by one
    for (int i = n - 1; i >= 0; i--) {
        swap(&arr[0], &arr[i]);
        heapify(arr, i, 0);
    }
}

int main() {
    int n = 1000;
    int* arr = (int*)malloc(n * sizeof(int));

    // Initialize array with random values
    srand(time(NULL));
    for (int i = 0; i < n; i++)
        arr[i] = rand() % 1000;

    heapSort(arr, n);

    free(arr);
    return 0;
}

File: Python/heapsort/llama-3.3-70b-versatile/llama-3.3-70b-versatile.py
def heapify(arr, n, i):
    """
    Heapify the array at index i.

    :param arr: The input array.
    :param n: The size of the heap.
    :param i: The index to heapify.
    """
    largest = i
    left = 2 * i + 1
    right = 2 * i + 2

    # Check if the left child is larger than the root.
    if left < n and arr[left] > arr[largest]:
        largest = left

    # Check if the right child is larger than the largest so far.
    if right < n and arr[right] > arr[largest]:
        largest = right

    # If the largest is not the root, swap and heapify the affected sub-tree.
    if largest != i:
        arr[i], arr[largest] = arr[largest], arr[i]
        heapify(arr, n, largest)


def heap_sort(arr):
    """
    Sort the array using heap sort.

    :param arr: The input array.
    """
    n = len(arr)

    # Build a max heap.
    for i in range(n // 2 - 1, -1, -1):
        heapify(arr, n, i)

    # Extract elements one by one.
    for i in range(n - 1, 0, -1):
        # Swap the root (max element) with the last element.
        arr[0], arr[i] = arr[i], arr[0]
        # Heapify the reduced heap.
        heapify(arr, i, 0)


import random

def main():
    arr = [random.randint(0, 10000) for _ in range(1000)]
    heap_sort(arr)

if __name__ == "__main__":
    main()